codec-http2/src/test/java/io/netty/handler/codec/http2/HpackDecoderTest.java

```diff
@@ -477,4 +480,34 @@ public void testDecodeCountsNamesOnlyOnce() throws Http2Exception {
             in.release();
         }
     }
+
+    @Test
+    public void testAccountForHeaderOverhead() throws Exception {
+        ByteBuf in = Unpooled.buffer(100);
+        try {
+            String headerName = "12345";
+            String headerValue = "56789";
+            long headerSize = headerName.length() + headerValue.length();
+            hpackDecoder.setMaxHeaderListSize(headerSize, 100);
+            HpackEncoder hpackEncoder = new HpackEncoder(true);
+
+            Http2Headers toEncode = new DefaultHttp2Headers();
+            toEncode.add(headerName, headerValue);
+            hpackEncoder.encodeHeaders(1, in, toEncode, NEVER_SENSITIVE);
+
+            Http2Headers decoded = new DefaultHttp2Headers();
+            try {
+                // SETTINGS_MAX_HEADER_LIST_SIZE is big enough for the header to fit...
+                assertThat(hpackDecoder.getMaxHeaderListSize(), is(greaterThanOrEqualTo(headerSize)));
+
+                hpackDecoder.decode(1, in, decoded);
+                fail();
+            } catch (Http2Exception e) {
```

Comment:
just change this to `catch (Http2Exception.HeaderListSizeException expected)` and remove the `assertTrue(...)` below ?